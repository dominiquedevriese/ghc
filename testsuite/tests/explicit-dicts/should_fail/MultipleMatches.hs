module MultipleMatches where

import GHC.Classes (Eq(..))

neverEq :: Eq.Dict a
neverEq = Eq.Dict { (==) = \_ _ -> False, (/=) = \_ _ -> True }

f :: (Eq a, Eq a) => a -> String
f = undefined

g = f ((neverEq :: Eq a))
