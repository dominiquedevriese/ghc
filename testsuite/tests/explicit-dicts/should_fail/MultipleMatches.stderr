
MultipleMatches.hs:11:5: error:
    • More than one constraint matches ‘Eq a’ in:
        forall a. (Eq a, Eq a) => a -> String
    • In the expression: f ((neverEq :: Eq a))
      In an equation for ‘g’: g = f ((neverEq :: Eq a))
